angular-ngrx-scss/src/app/repository/services/repository.service.ts

```diff
@@ -93,106 +66,88 @@ export class RepositoryService {
-          type: value.type,
-          path: value.path,
-        }));
-      }),
-    );
-  }
+    pullNumber: number,
+  ): Observable<IssueComments> {
+    const owner = encodeURIComponent(repoOwner);
+    const name = encodeURIComponent(repoName);
+    const pullId = encodeURIComponent(pullNumber);
+    const url = `${environment.githubUrl}/repos/${owner}/${name}/issues/${pullId}/comments`;
 
-  getFileContents(
-    owner: string,
-    repoName: string,
-    path: string,
-    commitOrBranchOrTagName: string,
-  ): Observable<FileContents> {
-    const url = `${environment.githubUrl}/repos/${owner}/${repoName}/contents/${path}?ref=${commitOrBranchOrTagName}`;
-    return this.http.get<FileContentsApiResponse>(url).pipe(
-      map((data) => {
-        return {
-          name: data.name,
-          type: data.type,
-          // TODO: consider using a function that also takes encoding format to decode this
-          content: atob(data.content),
-          size: data.size,
-        };
-      }),
-    );
+    return this.http.get<IssueComments>(url, {
+      headers: {
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
   }
 
+  /**
+   * NOTE: This call uses the search URL to find the information, and is a bit of a duplicate of other calls that use the repo URL. Both work fine and are provided currently.
+   * Gets a list of pull requests matching the provided state
+   * @param repoOwner who the repo belongs to
+   * @param repoName name of the repo
+   * @param prState if the pr is open or closed
+   * @returns the total count of state-matching pull requests and information for each of those pulls
+   */
   getPullRequests(
-    owner: string,
+    repoOwner: string,
     repoName: string,
     prState: PR_STATE,
-  ): Observable<RepoPullRequests> {
-    const url = `${environment.githubUrl}/search/issues?q=repo:${owner}/${repoName}+type:pr+state:${prState}`;
-    return this.http.get<PullRequestAPIResponse>(url).pipe(
-      map((data) => {
-        return {
-          totalCount: data.total_count,
-          pullRequests: data.items.map((item) => ({
-            id: item.id,
-            login: item.user.login,
-            title: item.title,
-            number: item.number,
-            state: item.state,
-            closedAt: item.closed_at ? new Date(item.closed_at) : null,
-            mergedAt: item.pull_request.merged_at
-              ? new Date(item.pull_request.merged_at)
-              : null,
-            createdAt: new Date(item.created_at),
-            labels: item.labels,
-            commentCount: item.comments,
-            labelCount: item.labels.length,
-          })),
-        };
-      }),
-    );
-  }
+  ): Observable<PullRequestAPIResponse> {
+    const owner = encodeURIComponent(repoOwner);
+    const name = encodeURIComponent(repoName);
+    const state = encodeURIComponent(prState);
+    const url = `${environment.githubUrl}/search/issues?q=repo:${owner}/${name}+type:pr+state:${state}`;
```

Comment:
an interesting idea just popped into my head to avoid having to explicitly encode all the params. We could use a tagged template to achieve the same thing with less boilerplate.
Just putting it as a discussion point and it definitely can be done on a separate ticket/PR

Reply:
What do you mean by tagged template? Would be curious to see an idea of what that might look like. Could then create an improvement ticket for that!

Reply:
have you created a ticket for that?

Reply:
Ticket here: https://github.com/thisdot/starter.dev-github-showcases/issues/824